### ISSUE TITLE
StackNavigator speaks accessibilityLabels of previous screens on iOS

### ISSUE THREAD

**Issue by sdg9** [react-navigation/react-navigation#377]
When using the StackNavigator accessibility labels are recognized even if they belong to the non-active scene.

Given the following setup
```javascript
const AppNavigator = StackNavigator({
  Home: {
    screen: MyHomeScreen,
  },
  Profile: {
    screen: MyProfileScreen,
  },
});

// boilerplate code...

render() {
    return (
      <AppNavigator />
    );
  }
```
If you navigate from Home to Profile accessibility elements from Home will be accessible on Profile.  This is also [a problem with Navigation Experimental](https://github.com/facebook/react-native/issues/9725) and has an [open PR](https://github.com/facebook/react-native/pull/11788) to address the issue in iOS.  After playing around with React Navigation I believe it is the same or extremely similar problem when using React Navigation's StackNavigator.

There is another [similar issue with React Navigation drawers](https://github.com/react-community/react-navigation/issues/311) but I created this separately as it is an issue using StackNavigator.

**Comment by kirankalyan5** [comment]
any workaround or updates on this ?

**Comment by sdg9** [comment]
I was doing [this](https://github.com/facebook/react-native/issues/9725#issuecomment-303104243) a while back but haven't looked at it recently, things may have changed.

**Comment by kelset** [comment]
Pinging OP @sdg9 since this issue is quite old and related to an old version of the lib. If no answer I'll close in 7 days.

**Comment by sdg9** [comment]
@kelset yeah it is still a problem but pretty sure react-navigation has a dependency on react-native before it can be solved.

This core issue started back with NavigatorExperimental #9725 and still exists now that react-navigation is the recommended approach.
React native doesn't have a good solution to iOS and Android parity around accessibility #11788

What I've done to manually solve this while waiting on #11788 is essentially fork both react-native and react-navigation.  

It's a small react-native change, captured in [this commit](https://github.com/sdg9/react-native/commit/a001149ccb6c071a020794861008970038612978) and also a small react-navigation change, described [in this comment](https://github.com/facebook/react-native/pull/11788#issuecomment-330617289).

Other people have been running into the same problem based on the chatter in #11788.

**Comment by kelset** [comment]
Hi @sdg9 thanks for the clarification; I guess then this needs to be open until the [`react-native` PR](https://github.com/facebook/react-native/pull/11788) gets reviewed an merged.

(btw it looks like there are some conflicts in the files)

**Comment by sdg9** [comment]
Yeah it's not my PR though, I just added input.  Unfortunately it's been relatively inactive since Feb 2017.

**Comment by kelset** [comment]
I'll try to bring it to the attention of the core team, but can't promise anything :/

**Comment by LucasdeCastro** [comment]
any workaround or updates on this ?

**Comment by Lily418** [comment]
@LucasdeCastro What version of React Navigation, React Native, React are you using? I just updated from 1.0.0-beta.19 to 1.0.0-beta.23 to try to fix this issue and very tentatively think it may have fixed this issue although I'm going to do some more testing to confirm that.

**Comment by brentvatne** [comment]
@Lily418 - did it resolve your issue?

**Comment by Lily418** [comment]
@brentvatne Yes, I think so

**Comment by brentvatne** [comment]
great. if anyone runs into this issue again please open a new issue :) thanks!
